packages/fxa-settings/src/pages/ResetPassword/ConfirmTotpResetPassword/index.tsx

```diff
@@ -0,0 +1,152 @@
+  verifyRecoveryCode: (code: string) => Promise<void>;
+  codeErrorMessage: string;
+  setCodeErrorMessage: Dispatch<SetStateAction<string>>;
+};
+
+const ConfirmTotpResetPassword = ({
+  verifyCode,
+  verifyRecoveryCode,
+  codeErrorMessage,
+  setCodeErrorMessage,
+}: ConfirmTotpResetPasswordProps) => {
+  const ftlMsgResolver = useFtlMsgResolver();
+  const [showRecoveryCode, setShowRecoveryCode] = useState<boolean>(false);
+
+  const totpFormAttributes: FormAttributes = {
+    inputFtlId: 'confirm-totp-reset-password-input-label',
+    inputLabelText: 'Enter code',
+    pattern: '[0-9]{6}',
+    maxLength: 6,
+    submitButtonFtlId: 'confirm-totp-reset-password-confirm-button',
+    submitButtonText: 'Confirm',
+  };
+
+  const recoveryCodeFormAttributes: FormAttributes = {
+    inputFtlId: 'confirm-recovery-code-reset-password-input-label',
+    inputLabelText: 'Enter 10-digit backup authentication code',
+    inputMode: InputModeEnum.text,
+    pattern: '[a-zA-Z0-9]',
+    maxLength: 10,
+    submitButtonFtlId: 'confirm-totp-reset-password-confirm-button',
+    submitButtonText: 'Confirm',
+  };
+
+  const localizedCustomCodeRequiredMessage = ftlMsgResolver.getMsg(
+    'signin-totp-code-required-error',
+    'Authentication code required'
+  );
+
+  return (
+    <div>
+      {showRecoveryCode ? (
+        <AppLayout>
+          <h1 className="text-grey-400 mb-6 text-left">
+            <FtlMsg id="confirm-totp-reset-password-header">
+              Reset your password
+            </FtlMsg>
+          </h1>
+
+          <h2 className="font-bold text-xl text-left">
+            <FtlMsg id="confirm-totp-reset-password-subheader">
+              Enter your backup recovery code
+            </FtlMsg>
+          </h2>
+
+          <div className="flex space-x-4">
+            <img src={protectionShieldIcon} alt="" />
+            <FtlMsg id="confirm-totp-reset-password-instruction">
+              <p id="totp-code-instruction" className="my-5 text-md text-left">
+                Check your download or saved backup recovery code.
+              </p>
+            </FtlMsg>
+          </div>
+
+          <FormVerifyCode
+            {...{
+              formAttributes: recoveryCodeFormAttributes,
+              verifyCode: verifyRecoveryCode,
+              localizedCustomCodeRequiredMessage,
+              viewName: 'confirm-recovery-code-reset-password',
+              codeErrorMessage,
+              setCodeErrorMessage,
+            }}
+          />
+          <div className="mt-5 link-blue text-sm flex justify-end">
+            <FtlMsg id="confirm-recovery-code-reset-password-trouble-code">
+              <div
+                onClick={() => {
+                  setShowRecoveryCode(false);
+                }}
+              >
+                Back
+              </div>
+            </FtlMsg>
+          </div>
+        </AppLayout>
+      ) : (
+        <AppLayout>
+          <h1 className="text-grey-400 mb-6 text-left">
```

Comment:
Nit that we could use `base-class` for `AppLayout`, but we will definitely want to not use `text-left` and instead use `text-start` for all of these instances, otherwise rtl will not display this well.